File: Broadcast/MyBroadcastReceiver/java/app/src/main/java/com/dicoding/picodiploma/mybroadcastreceiver/SmsReceiverActivity.java
package com.dicoding.picodiploma.mybroadcastreceiver;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.dicoding.picodiploma.mybroadcastreceiver.databinding.ActivitySmsReceiverBinding;

public class SmsReceiverActivity extends AppCompatActivity {

    public static final String EXTRA_SMS_NO = "extra_sms_no";
    public static final String EXTRA_SMS_MESSAGE = "extra_sms_message";

    private ActivitySmsReceiverBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivitySmsReceiverBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setTitle(getString(R.string.incoming_message));

        binding.btnClose.setOnClickListener(v -> finish());

        String senderNo = getIntent().getStringExtra(EXTRA_SMS_NO);
        String senderMessage = getIntent().getStringExtra(EXTRA_SMS_MESSAGE);

        binding.tvFrom.setText(String.format("from : %s", senderNo));
        binding.tvMessage.setText(senderMessage);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        binding = null;
    }
}

File: Broadcast/MyBroadcastReceiver/kotlin/app/src/main/java/com/dicoding/picodiploma/mybroadcastreceiver/SmsReceiverActivity.kt
package com.dicoding.picodiploma.mybroadcastreceiver

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_sms_receiver.*

class SmsReceiverActivity : AppCompatActivity(), View.OnClickListener {

    companion object {
        const val EXTRA_SMS_NO = "extra_sms_no"
        const val EXTRA_SMS_MESSAGE = "extra_sms_message"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sms_receiver)

        title = getString(R.string.incoming_message)

        btn_close.setOnClickListener(this)

        val senderNo = intent.getStringExtra(EXTRA_SMS_NO)
        val senderMessage = intent.getStringExtra(EXTRA_SMS_MESSAGE)

        tv_from.text = getString(R.string.from, senderNo)
        tv_message.text = senderMessage
    }

    override fun onClick(v: View) {
        if (v.id == R.id.btn_close) {
            finish()
        }
    }
}
